### PR TITLE
chore(fmt/linters): enable nilerr, nilnil, nolintlint and fix related issues

### PR DESCRIPTION
**Pull request by walldiss**
Enables  nilerr, nilnil, nolintlint checks as well as fixing issues found by those linters

### CODE REVIEW

File: state/core_access.go [hunk 1]
```diff
@@ -604,7 +604,7 @@ func (ca *CoreAccessor) GrantFee(
 
 	msg, err := feegrant.NewMsgGrantAllowance(allowance, granter, grantee)
 	if err != nil {
-		return nil, nil
+		return nil, err
```
Comment (Wondertan) [inline state/core_access.go:607]: Nice, actual bur

File: core/eds.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 // nil is returned in place of the eds.
 func extendBlock(data types.Data, appVersion uint64, options ...nmt.Option) (*rsmt2d.ExtendedDataSquare, error) {
 	if app.IsEmptyBlock(data, appVersion) {
-		return nil, nil
+		return share.EmptyExtendedDataSquare(), nil
```
Comment (Wondertan) [inline core/eds.go:31]: This might introduce the bug, pls check how the code is being used

Comment (walldiss) [reply core/eds.go:31]: I agree. There seem to be no special handling for nil eds further in the code. Need @renaynay to confirm

Comment (Wondertan) [reply core/eds.go:31]: All the tests are broken, so likely related

Comment (walldiss) [reply core/eds.go:31]: Just flack, unrelated. 
```
discovery_test.go:134: did not discover peer in time
```
https://github.com/celestiaorg/celestia-node/actions/runs/8783227354/job/24099029082?pr=3332#step:4:87

Comment (walldiss) [reply core/eds.go:31]: https://github.com/celestiaorg/celestia-node/pull/3332/commits/d97a9d359bc178677c758e0b34354b418af02f23

**Review by Wondertan — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**